File: 0879-maximize-distance-to-closest-person/solution.cpp
class Solution {
public:
    int maxDistToClosest(vector<int>& seats) {
        const int n = seats.size();
        int res = 0 , l = -1;
        for(int i = 0 ; i < n ; i++){
            if(seats[i] == 1){
                res = l < 0 ? i : max(res, (i-l)/2) ;
                l = i ;
            }
        }
        
        return max(res,n-1-l);
    }
};

File: 0879-maximize-distance-to-closest-person/solution.java
class Solution {
    public int maxDistToClosest(int[] seats) {
        
    int n = seats.length;
    int res = 0 , l = -1 ;
    
    for(int i = 0 ; i < n ; i++){
        
        if(seats[i] == 1){
            res = l < 0 ? i : Math.max(res , (i-l)/2);
            l = i;
        }
        
    }
    return Math.max(res,n-1-l);
    }
}
